ImperatorToCK3/Source/Imperator/Countries/CountryName.cpp

```diff
@@ -1,20 +1,118 @@
 #include "CountryName.h"
+#include "Country.h"
+#include <ranges>
 #include "Log.h"
-#include "ParserHelpers.h"
-#include "CommonRegexes.h"
 
 
 
-Imperator::CountryName::CountryName(std::istream& theStream) {
-	registerKeys();
-	parseStream(theStream);
-	clearRegisteredKeywords();
+template<class T>
+std::unique_ptr<T> copy_unique(const std::unique_ptr<T>& source) {
```

Comment:
If you're copying unique pointers, why not just use shared ones?

Reply:
There's no need to, the bases aren't shared between instances. I'm just copying the value.